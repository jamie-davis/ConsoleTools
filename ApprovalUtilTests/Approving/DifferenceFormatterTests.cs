using ApprovalUtil.Approving;
using ConsoleToolkit.ConsoleIO;
using ConsoleToolkit.Testing;
using TestConsoleLib.Testing;

namespace ApprovalUtilTests.Approving;

public class DifferenceFormatterTests
{
    [Fact]
    public void DifferencesAreFormatted()
    {
        //Arrange
        var testConsole = new UnitTestConsole(typeof(Program).Namespace);
        var console = testConsole.Console;
        var error = testConsole.Error;

        var received = "The received file";
        var approved = "The approved file";

        //Act
        DifferenceFormatter.DisplayDiffs(received, approved, console, error);

        //Assert
        testConsole.Interface.GetBuffer(ConsoleBufferFormat.TextOnly).Verify();
    }
}