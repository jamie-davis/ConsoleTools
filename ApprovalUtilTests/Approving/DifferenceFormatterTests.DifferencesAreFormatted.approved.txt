=================                                           
>> ---                                                      
The路approved路file                                           
~~~~~~~~~~~~~~~~~~~~                                        
>> +++                                                      
The路received路file                                           
=================                                           
                                                            